Core/src/RootExt.cpp

```diff
@@ -84,6 +86,52 @@ ClassInheritance FindClassInheritance(const std::string& class_name)
 
     return inheritance;
 }
+
+void RebinAndFill(TH2& new_hist, const TH2& old_hist)
+{
+    static const auto check_range = [](const TAxis* old_axis, const TAxis* new_axis) {
+        const double old_min = old_axis->GetBinLowEdge(1);
+        const double old_max = old_axis->GetBinUpEdge(old_axis->GetNbins());
+        const double new_min = new_axis->GetBinLowEdge(1);
+        const double new_max = new_axis->GetBinUpEdge(new_axis->GetNbins());
+        return old_min <= new_min && old_max >= new_max;
+    };
+
+    static const auto get_new_bin = [](const TAxis* old_axis, const TAxis* new_axis, int bin_id_old) {
+        const double old_low_edge = old_axis->GetBinLowEdge(bin_id_old);
+        const double old_up_edge = old_axis->GetBinUpEdge(bin_id_old);
+        const int bin_low_new = new_axis->FindFixBin(old_low_edge);
+        const int bin_up_new = new_axis->FindFixBin(old_up_edge);
+
+        if(bin_low_new != bin_up_new
+          && !(fabs(old_up_edge-new_axis->GetBinUpEdge(bin_low_new)) <= std::numeric_limits<double>::epsilon()*2))
+          throw analysis::exception("Uncompatible bin edges");
```

Comment:
Few comments about this line:
1) e.g: lets say we have the following bins: 0.0, 1.0, 2.0... in the situation when we have old_low_edge= 0.8 and old_up_edge=1.0, `new_axis->FindFixBin(1.0)` will return 1, but `new_axis->FindFixBin(1.0)` will return 2, I do not know why but this method include edge in the next bin.... 
2) in this case we need to check if upper limit of the bin 1 corresponds to old_up_edge... but that does not work because `GetBinUpEdge(1)` returns it with numeric type inaccuracy 1.00000000000000009..... `old_axis->GetBinUpEdge(bin_id_old)` also was returned with numerical inaccuracy, so the difference should be within `numeric_limits<double>::epsilon() * 2` (it is interesting that difference is alwaysexactly two epsolons)

Reply:
- If bins are defined explicitly (as an array) there is no numeric inaccuracy involved because there were no floating-point operations applied to the number. I don't understand why for input bins = { 0., 1., 2. }, GetBinUpEdge(1) would return something different from 1.
- Indeed, all bin intervals in the TH histograms are in the form [a, b), i.e. lower bound is included, but the upper bound is not. I forgot to account for it in the draft. This additional condition should solve that. However, epsilon is a relative precision, so I propose to check the compatibility in the following way:
```c++
const double new_up_edge = old_axis->GetBinUpEdge(bin_low_new);
if(bin_low_new != bin_up_new
        && !(std::abs(old_up_edge-new_up_edge) <= std::numeric_limits<double>::epsilon() * std::abs(old_up_edge+new_up_edge) * 2))
    throw analysis::exception("Uncompatible bin edges");
```